addOns/automation/src/main/java/org/zaproxy/addon/automation/gui/ActiveScanJobDialog.java

```diff
@@ -0,0 +1,393 @@
+        }
+
+        this.addComboField(1, DEFAULT_THRESHOLD_PARAM, allthresholds, thresholdName);
+
+        List<String> allstrengths = new ArrayList<>();
+
+        for (AttackStrength at : AttackStrength.values()) {
+            if (AttackStrength.DEFAULT.equals(at)) {
+                continue;
+            }
+            allstrengths.add(JobUtils.strengthToI18n(at.name()));
+        }
+
+        this.addComboField(1, DEFAULT_STRENGTH_PARAM, allstrengths, strengthName);
+
+        this.addPadding(1);
+
+        List<JButton> buttons = new ArrayList<>();
+        buttons.add(getAddButton());
+        buttons.add(getModifyButton());
+        buttons.add(getRemoveButton());
+
+        this.addTableField(2, getRulesTable(), buttons);
+
+        this.addNumberField(
+                3,
+                DELAY_IN_MS_PARAM,
+                0,
+                Integer.MAX_VALUE,
+                JobUtils.unBox(JobUtils.unBox(job.getParameters().getDelayInMs())));
+        this.addNumberField(
+                3,
+                THREADS_PER_HOST_PARAM,
+                0,
+                Integer.MAX_VALUE,
+                JobUtils.unBox(JobUtils.unBox(job.getParameters().getThreadPerHost())));
+        this.addCheckBoxField(
+                3, ADD_QUERY_PARAM, JobUtils.unBox(job.getParameters().getAddQueryParam()));
+        this.addCheckBoxField(
+                3,
+                HANDLE_ANTI_CSRF_PARAM,
+                JobUtils.unBox(job.getParameters().getHandleAntiCSRFTokens()));
+        this.addCheckBoxField(
+                3,
+                INJECT_PLUGIN_ID_PARAM,
+                JobUtils.unBox(job.getParameters().getInjectPluginIdInHeader()));
+        this.addCheckBoxField(
+                3,
+                SCAN_HEADERS_PARAM,
+                JobUtils.unBox(job.getParameters().getScanHeadersAllRequests()));
+
+        this.addPadding(3);
+
+        setAdvancedTabs(getBoolValue(FIELD_ADVANCED));
+    }
+
+    private boolean advOptionsSet() {
+        Parameters params = this.job.getParameters();
+        return params.getDelayInMs() != null
+                || params.getThreadPerHost() != null
+                || params.getAddQueryParam() != null
+                || params.getHandleAntiCSRFTokens() != null
+                || params.getInjectPluginIdInHeader() != null
+                || params.getScanHeadersAllRequests() != null;
+    }
+
+    private void setAdvancedTabs(boolean visible) {
+        // Show/hide the advanced tab
+        this.setTabsVisible(new String[] {"automation.dialog.ascan.tab.adv"}, visible);
+    }
+
+    @Override
+    public void save() {
+        this.job.getData().setName(this.getStringValue(NAME_PARAM));
+        this.job.getParameters().setContext(this.getStringValue(CONTEXT_PARAM));
+        this.job.getParameters().setPolicy(this.getStringValue(POLICY_PARAM));
+        // TODO this.job.getParameters().setUrl(this.getStringValue(URL_PARAM));
```

Comment:
Still pending?

Reply:
yeah, but I'll remove the commented code as I have this noted...